projects/mui-angular/button/button.component.ts

```diff
@@ -1,36 +1,82 @@
-import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
-
+import { Component, ElementRef, Input, ViewChild, HostListener, Renderer2 } from '@angular/core';
 
 @Component({
   selector: 'mui-button',
   template: `
-    <button class="mui-btn" #button>
+    <button class="mui-btn" #button [disabled]="disabled">
       <ng-content></ng-content>
+      <span class="mui-btn__ripple-container" #container><span class="mui-ripple" #ripple></span></span>
     </button>
   `
 })
 
-
-export class ButtonComponent implements OnInit {
+export class ButtonComponent {
   @ViewChild('button') button: ElementRef;
+  @ViewChild('ripple') ripple: ElementRef;
 
   @Input() variant: string = '';
   @Input() color: string = '';
   @Input() size: string = '';
-  
-  constructor() {
+  @Input() disabled?: boolean = false;
+
+  @HostListener('mousedown', ['$event', '$event.currentTarget'])
+  mouseDown(event: MouseEvent) {
+    if(this.disabled) return;
+    if (event.button === 0) {
+      this.getFadeOutAnimation(event);
+    }
+  }
+
+  @HostListener('mouseup', ['$event', '$event.currentTarget']) mouseUp() {
+    this.getFadeInAnimation()
   }
 
-  ngOnInit(): void {
+  constructor(private renderer: Renderer2) {
   }
 
   ngAfterViewInit() {
     // set button styles
     ['variant', 'color', 'size'].forEach(attrName => {
       let attrVal = this[attrName];
       if (attrVal) {
-        this.button.nativeElement.classList.add('mui-btn--' + attrVal);
+        this.renderer.addClass(this.button.nativeElement, 'mui-btn--' + attrVal);
       }
     });
   }
+
+  private getFadeOutAnimation(event) {
+    var offsetWidth, offsetHeight, offsetTop, offsetLeft, radius, diameter, width, height, left, top;
+
+    // get (x, y) position of click
+    offsetWidth = this.button.nativeElement.offsetWidth;
```

Comment:
You can save some keystrokes by saving references to `this.button.nativeElement` and `this.ripple.nativeElement`

Reply:
added private variables and set in ngAfterViewInit